src/displayapp/screens/StopWatch.cpp

```diff
@@ -123,96 +144,89 @@ void StopWatch::SetInterfaceStopped() {
+    blinkTime = currentTime + blinkInterval;
+    if (lv_obj_get_state(time, LV_LABEL_PART_MAIN) == LV_STATE_DEFAULT) {
+      lv_obj_set_state(time, LV_STATE_DISABLED);
+      lv_obj_set_state(msecTime, LV_STATE_DISABLED);
+    } else {
+      lv_obj_set_state(time, LV_STATE_DEFAULT);
+      lv_obj_set_state(msecTime, LV_STATE_DEFAULT);
+    }
+  }
 }
 
-void StopWatch::Refresh() {
-  if (currentState == States::Running) {
-    laps[lapsDone] = oldTimeElapsed + xTaskGetTickCount() - startTime;
+void StopWatch::RenderLaps() {
+  lv_label_set_text(lapText, "");
+  for (int i = 0; i < displayedLaps; i++) {
+    std::optional<LapInfo> lap = stopWatchController.GetLapFromHistory(i);
 
-    TimeSeparated_t currentTimeSeparated = convertTicksToTimeSegments(laps[lapsDone]);
-    if (currentTimeSeparated.hours == 0) {
-      lv_label_set_text_fmt(time, "%02d:%02d", currentTimeSeparated.mins, currentTimeSeparated.secs);
+    if (lap) {
+      TimeSeparated laptime = ConvertTicksToTimeSegments(lap->timeSinceStart);
+      char buffer[16];
+      sprintf(buffer, "#%2d   %2d:%02d.%02d\n", lap->number, laptime.mins, laptime.secs, laptime.hundredths);
+      lv_label_ins_text(lapText, LV_LABEL_POS_LAST, buffer);
     } else {
-      lv_label_set_text_fmt(time, "%02d:%02d:%02d", currentTimeSeparated.hours, currentTimeSeparated.mins, currentTimeSeparated.secs);
-      if (!isHoursLabelUpdated) {
-        lv_obj_set_style_local_text_font(time, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, &jetbrains_mono_42);
-        lv_obj_realign(time);
-        isHoursLabelUpdated = true;
-      }
-    }
-    lv_label_set_text_fmt(msecTime, "%02d", currentTimeSeparated.hundredths);
-  } else if (currentState == States::Halted) {
-    const TickType_t currentTime = xTaskGetTickCount();
-    if (currentTime > blinkTime) {
-      blinkTime = currentTime + blinkInterval;
-      if (lv_obj_get_state(time, LV_LABEL_PART_MAIN) == LV_STATE_DEFAULT) {
-        lv_obj_set_state(time, LV_STATE_DISABLED);
-        lv_obj_set_state(msecTime, LV_STATE_DISABLED);
-      } else {
-        lv_obj_set_state(time, LV_STATE_DEFAULT);
-        lv_obj_set_state(msecTime, LV_STATE_DEFAULT);
-      }
+      lv_label_ins_text(lapText, LV_LABEL_POS_LAST, "\n");
     }
   }
 }
 
-void StopWatch::playPauseBtnEventHandler() {
-  if (currentState == States::Init || currentState == States::Halted) {
-    Start();
-  } else if (currentState == States::Running) {
-    Pause();
+void StopWatch::Refresh() {
+  if (stopWatchController.IsRunning()) {
+    RenderTime();
+  } else if (stopWatchController.IsPaused()) {
+    RenderPause();
+  }
+}
+
+void StopWatch::PlayPauseBtnEventHandler() {
+  if (stopWatchController.IsCleared() || stopWatchController.IsPaused()) {
+    stopWatchController.Start();
+    DisplayStarted();
+    wakeLock.Lock();
+  } else if (stopWatchController.IsRunning()) {
+    stopWatchController.Pause();
+    blinkTime = xTaskGetTickCount() + blinkInterval;
+    DisplayPaused();
+    RenderTime();
```

Comment:
Maybe rendertime before displaypaused? Rendering the buttons takes a while and there is a noticeable delay for the time to update on my watch

Reply:
Applied in `OnPause()`.